app/api/snippets/[id]/comments/route.ts

```diff
@@ -0,0 +1,74 @@
+import { NextRequest, NextResponse } from "next/server";
+import dbConnect from "@/lib/db/connect";
+import Snippet from "@/lib/db/snippetModel";
+import Comment from "@/lib/db/commentModel";
+import { currentUser } from "@clerk/nextjs/server";
+import { Types } from "mongoose";
+
+export async function POST(
+  req: NextRequest,
+  { params }: { params: { id: string } }
+) {
+  try {
+    await dbConnect();
+    const { content } = await req.json();
+    const snippetId = params.id;
+
+    const user = await currentUser();
+    const userId = user?.id;
+
+    if (!userId) {
+      return NextResponse.json(
+        { error: "User not authenticated" },
+        { status: 401 }
+      );
+    }
+
+    const snippet = await Snippet.findById(snippetId);
+    if (!snippet) {
+      return NextResponse.json({ error: "Snippet not found" }, { status: 404 });
+    }
+
+    const newComment = new Comment({
+      snippetId,
+      userId,
+      content,
+    });
+    await newComment.save();
+
+    snippet.comments = snippet.comments || [];
+    snippet.comments.push(newComment._id as Types.ObjectId);
+    await snippet.save();
```

Comment:
and when you are saving the comment using the snippet model it is asking for the required field i.e category and all, which is giving the error.